### PR TITLE
Implements OAuth Node support

### PR DESCRIPTION
**Pull request by skansal-plaid**
- The quickstart already supports the OAuth desktop flow even before this PR. 
- This PR adds support for the OAuth redirect flow (used for mobile web).
- This PR does not add support for the OAuth redirect + webview flow (used for mobile apps).

I tested that this PR works locally for the OAuth flows and does not break the existing classic Plaid Quickstart flow.

Note: The quickstart does not currently demonstrate the webview flow. Handling the webview flow requires defining URLs at the quickstart app to handle Link success and error (https://plaid.com/docs/#communication-between-link-and-your-app). This PR does not add webview support.

### CODE REVIEW

File: node/README.md [hunk 1]
```diff
@@ -22,3 +22,20 @@ PLAID_COUNTRY_CODES='US,CA,GB,FR,ES' \
...
+# You will need to whitelist the OAuth redirect URI for your client ID through
+# the Plaid developer dashboard at https://dashboard.plaid.com/team/api.
+PLAID_CLIENT_ID='CLIENT_ID' \
+PLAID_SECRET='SECRET' \
+PLAID_PUBLIC_KEY='PUBLIC_KEY' \
+PLAID_ENV='sandbox' \
+PLAID_PRODUCTS='transactions' \
+PLAID_COUNTRY_CODES='GB' \
+PLAID_OAUTH_REDIRECT_URI='http://localhost:8000/oauth-response.html' \
+PLAID_OAUTH_NONCE='nice-and-long-nonce' \
```
Comment (notthefakestephen) [inline node/README.md]: do you think we should provide instructions on how to generate a nonce here?

Comment (jacksiegel) [reply node/README.md]: @notthefakestephen did you see https://github.com/plaid/quickstart/pull/91#discussion_r373408747?

File: node/README.md [hunk 2]
```diff
@@ -22,3 +22,20 @@ PLAID_COUNTRY_CODES='US,CA,GB,FR,ES' \
 node index.js
 # Go to http://localhost:8000
 ```
+
+``` bash
```
[the remark this answers is not in the document]
Comment (skansal-plaid) [reply node/README.md]: https://github.com/plaid/quickstart/pull/91/commits/70416ee0e019e82463f1ccde96025f16f76262e6

[the remark this answers is not in the document]
Comment (skansal-plaid) [reply node/README.md]: https://github.com/plaid/quickstart/pull/91/commits/0e03c4e500e0738255d703404eb5f49bbacda2b4

File: node/index.js [hunk 3]
```diff
@@ -23,6 +22,16 @@ var PLAID_PRODUCTS = envvar.string('PLAID_PRODUCTS', 'transactions');
...
 
+// Parameters used for the OAuth redirect Link flow.
+//
+// Set PLAID_OAUTH_REDIRECT_URI to 'http://localhost:8000/oauth-response.html'
+// The OAuth redirect flow requires an end-point on the developer's website
+// that the bank website should redirect to. You will need to whitelist
+// this redirect URI for your client ID through the Plaid developer dashboard
+// at https://dashboard.plaid.com/team/api.
+var PLAID_OAUTH_REDIRECT_URI = envvar.string('PLAID_OAUTH_REDIRECT_URI', '');
+// Set PLAID_OAUTH_NONCE to 'nice-and-long-nonce'
```
[the remark this answers is not in the document]
Comment (skansal-plaid) [reply node/index.js]: thanks

File: node/views/oauth-response.ejs [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+          location.href = 'http://localhost:8000';
+        });
+      },
+    });
+    handler.open();
+
+  })(jQuery);
+
+  function qs(key) {
+    key = key.replace(/[*+?^$.\[\]{}()|\\\/]/g, "\\$&"); // escape RegEx meta chars
```
[the remark this answers is not in the document]
Comment (skansal-plaid) [reply node/views/oauth-response.ejs]: tested locally after removing this and the oauth flow still works

File: node/README.md [hunk 5]
```diff
@@ -1,24 +1,60 @@
...
+#
+# PLAID_PRODUCTS is a comma-separated list of products to use when
+# initializing Link.
+#
+# You will need to whitelist the PLAID_OAUTH_REDIRECT_URI for
+# your client ID through the Plaid developer dashboard at
+# https://dashboard.plaid.com/team/api.
+#
+# Set PLAID_OAUTH_NONCE to a unique identifier such as a UUID.
+# The nonce must be at least 16 characters long.
```
Comment (jacksiegel) [inline node/README.md]: I think the README would flow more naturally if the `PLAID_OAUTH_REDIRECT_URI` and `PLAID_OAUTH_NONCE` comments were moved up to the `OAuth redirect flow` section above.

Comment (skansal-plaid) [reply node/README.md]: https://github.com/plaid/quickstart/pull/91/commits/657f5715332db729de91d59b5b1df36e84dfa7ff

File: node/views/index.ejs [hunk 6]
```diff
@@ -275,12 +302,7 @@
         // payment token to be generated before the Link handler can be
         // initialized.
         handler = Plaid.create({
-          apiVersion: 'v2',
-          clientName: 'Plaid Quickstart',
-          env: '<%= PLAID_ENV %>',
-          product: products,
-          key: '<%= PLAID_PUBLIC_KEY %>',
-          countryCodes: '<%= PLAID_COUNTRY_CODES %>'.split(','),
+          ...linkHandlerCommonOptions,
```
Comment (jasonjia93) [inline node/views/index.ejs]: I might have asked this before, but why are we making a request to `get_access_token` here if it isn't relevant?

Comment (skansal-plaid) [reply node/views/index.ejs]: The access token and item ID are displayed on the quickstart success page so this request it just to ensure the quickstart success pane still looks good.

**Pull request merged**